### ISSUE TITLE
Imax clone 80W up to 200W.

### ISSUE THREAD

**Issue by ElitePlusPlus** [stawel/cheali-charger#306]
Hi! I have upgrade hardware, but can't up charge/discharge current. I try all firmware, but only one  is works correctly. I change HardwareConfig and now can set change current up to 15A on 2V (I need to charge NiCd cell 15-110Ah), but when correct error I charge 14/I discharge 14 max current must be set up to 4,5A... Original firmware allowed set current up to 6A... Its good, but too low for my needs. What I do wrong and can I do something?

**Comment by diggit** [comment]
Hi, can you share, what kind of mods you did?
Error 14 means your limit is our of measurable range IIRC.

**Comment by ElitePlusPlus** [comment]
Hi! THX a lot for answer!
I change all transistor and diodes to more powerfull - IRF4905, IRF3205 and MBR20100. Then I made new coil and change resistor in the charge and discharge circuits for 5W once. I use two resistors 0,05 Om in parallel for current and two 0.1 Om in discharge. And now I can charge up to 10A, but reading device is less than twice. I want to add two resistors more and reach 20 A charging current...
![1](https://user-images.githubusercontent.com/87881347/128646415-05f02222-6a6a-4e5d-a980-910750ae8ea5.jpg)
![2](https://user-images.githubusercontent.com/87881347/128646416-27a8c73f-87c9-4e85-b4bd-ece245abc790.jpg)
![3](https://user-images.githubusercontent.com/87881347/128646417-5ef94a98-dcce-4ee4-8273-a5835142b311.jpg)
![4](https://user-images.githubusercontent.com/87881347/128646418-56754f63-e9c0-420d-a95d-1607540ae8f2.jpg)

**Comment by ElitePlusPlus** [comment]
And charging process...
![5](https://user-images.githubusercontent.com/87881347/128646766-e0a66814-7c03-4263-bbb9-f33154ec37ac.jpg)
![6](https://user-images.githubusercontent.com/87881347/128646769-11c6635d-2669-4d0f-b1cf-942a5da5a631.jpg)

**Comment by diggit** [comment]
So generally your Rcharge shunt is half of default value which should double Icharge range (good).
I am  not entirely sure about difference between `Ismps` and `IsmpsSet` here https://github.com/stawel/cheali-charger/blob/55aca4c162b46b770a4d29e8c0beb1c0f934f5a4/src/core/calibration/Calibration.cpp#L115
but `IsmpsSet` check fails.
I guess, that your charger can't set output so high, that it would reach desired output current.

Can you share your Icharge calibration values? (1 pair per calibration point)

**Comment by ElitePlusPlus** [comment]
1500 mA (3000 mA really) - 17646
300 mA (600 mA) - 1987

**Comment by diggit** [comment]
>1500 mA (**3000 mA really**) - 17646

What do you mean?
If calibration point is 1.5 A, you have to adjust your calibration so you get 1.5 A on output and not 3 A.

**Comment by ElitePlusPlus** [comment]
Yes, when I calibrate charger show 1500mA, then ammetr show 3000mA. I have resistor smaller then original - 0.025 Om.

**Comment by ElitePlusPlus** [comment]
Ok! I recalibrate to 1500 - 1500. And max charge current reach 15A. But discharge current max 3800mA on 1,2V... I will try to change resistor to 2x0,05 Om.

**Comment by ElitePlusPlus** [comment]
Ok!!! I made changes that I planed! Now I have near 6300 discharge current, but cell was not fully charged. Lets charge it.
![7](https://user-images.githubusercontent.com/87881347/130331949-5c271703-3b68-4f07-b2dd-68bca93458d2.jpg)
![8](https://user-images.githubusercontent.com/87881347/130331953-baa3ef74-3099-4232-a77a-838d0f647cd4.jpg)

**Comment by ElitePlusPlus** [comment]
And now I want to change firmware for chardge current up to 30A.

**Comment by ElitePlusPlus** [comment]
Big cells are waiting!!!
![9](https://user-images.githubusercontent.com/87881347/130332046-e4187865-1f01-4a11-9bac-c79ed80528f0.jpg)

**Comment by ElitePlusPlus** [comment]
Can anybody make some changes to firmvare? Need to add tripple charging algorithm for Ni-Cd cells...

**Comment by diggit** [comment]
What do you mean by "tripple charging algorithm"?
Also flash of Atmega32 if filled almost to the brim.

**Comment by ElitePlusPlus** [comment]
It is second stage of charge when after chardging 1C add 1/10C for few hours.
Maybe cut Pb charge?

**Comment by ElitePlusPlus** [comment]
Hi all! I made a new one of power PCB.
![10](https://user-images.githubusercontent.com/87881347/144712484-4b2165ca-01d6-4b75-849c-88c15cfc6199.jpg)
![11](https://user-images.githubusercontent.com/87881347/144712487-76a58164-491b-464f-ad30-8361f92e46f7.jpg)
 And I hope to reach dis/charging current up to 30A. Lets try...

**Comment by JFingerle** [comment]
Offtopic: Really impressive hardware mod!
